### PR TITLE
Fix mixed precision align issue after apply convert precision to graph which has SubGraphOp

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
 *Fix the issue in convert precision(FP32->FP16) with the graph has SubGraphOp, need convert precision if the SubGraphOp has different precision when set `keep_precision_sensitive_in_fp32 = true`*

-- jira CVS-128993

### CODE REVIEW

File: src/common/transformations/tests/utils/convert_precision.cpp [hunk 1]
```diff
@@ -2442,4 +2442,107 @@ TEST(TransformationTests, ConvertPrecision_assign_read_value_change_variable_typ
...
+
+        type_to_fuse_map empty_type_to_fuse_map = {};
+        bool keep_precision_sensitive_in_fp32 = true;
+        bool convert_input_output_precision = false;
+        manager.register_pass<pass::ConvertPrecision>(precisions_map{{element::f32, element::f16}},
+                                                      empty_type_to_fuse_map,
+                                                      keep_precision_sensitive_in_fp32,
+                                                      convert_input_output_precision);
+        ASSERT_NO_THROW(manager.run_passes(f));
+    }
```
Comment (songbell) [inline src/common/transformations/tests/utils/convert_precision.cpp:2547]: is the no throw check enough here? do we need to check that precision in subgraph is preserved or be converted as expected?

**Comment by iefode** [comment]
@itikhono Please take a look

File: src/common/transformations/tests/utils/convert_precision.cpp (hunk 1 above)
Comment (WeldonWangwang) [reply src/common/transformations/tests/utils/convert_precision.cpp:2547]: Update, thanks.

**Comment by songbell** [comment]
can you update desc to be more specific?
f32->f16 conversion generally is converted, exception is that we set keep_precision_sensitive_in_fp32 and the node itself is precision sensitive, why subgraph need to be precision sensitive in your change?

**Comment by WeldonWangwang** [comment]
> can you update desc to be more specific? f32->f16 conversion generally is converted, exception is that we set keep_precision_sensitive_in_fp32 and the node itself is precision sensitive, why subgraph need to be precision sensitive in your change?
image

The result of subgraph should keep same precision with the input of the output of subgraph, for example, if we set `keep_precision_sensitive_in_fp32 = true`, TestSubgraph_188 should keep the same precision with Parameter_180. But if there is EXP in subgraph, the subgraph's output will keep FP32, but the precision of Parameter_180 is FP16, it will fail.

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 2]
```diff
@@ -202,7 +202,9 @@ bool convert_node_input_precision(const std::shared_ptr<ov::Node>& node,
     return false;
 }
 
-bool convert_function_precision(const std::shared_ptr<Model>& f,
+bool convert_function_precision(ov::pass::PassBase& pass,
+                                std::shared_ptr<ov::pass::PassConfig> pass_config,
```
Comment (songbell) [inline src/common/transformations/src/transformations/convert_precision.cpp:206]: pass_config can be obtained by pass.get_pass_config()?

**Comment by songbell** [comment]
> > can you update desc to be more specific? f32->f16 conversion generally is converted, exception is that we set keep_precision_sensitive_in_fp32 and the node itself is precision sensitive, why subgraph need to be precision sensitive in your change?
> > image
> 
> The result of subgraph should keep same precision with the input of the output of subgraph, for example, if we set `keep_precision_sensitive_in_fp32 = true`, TestSubgraph_188 should keep the same precision with Parameter_180. But if there is EXP in subgraph, the subgraph's output will keep FP32, but the precision of Parameter_180 is FP16, it will fail.

so with your PR , what will be the graph like with fp32->fp16 transformation and keep_precision_sensitive_in_fp32 enabled?  matmul will be tagged with fp32 or fp16?

**Review by songbell — APPROVED** [review]

File: src/common/transformations/src/transformations/fp16_compression/align_mixed_fp32_fp16_types.cpp [hunk 3]
```diff
@@ -66,7 +66,7 @@ bool ov::pass::AlignMixedFP32FP16Types::run_on_model(const std::shared_ptr<ov::M
                         continue;
 
                     // todo xxx-101766: if we don't skip Results there is an error on GPU
-                    if (std::dynamic_pointer_cast<ov::op::v0::Result>(out_node))
+                    if (std::dynamic_pointer_cast<ov::op::v0::Result>(out_node) && !m_is_subgraph)
```
Comment (songbell) [inline src/common/transformations/src/transformations/fp16_compression/align_mixed_fp32_fp16_types.cpp:69]: maybe we can enable a test to below as well: src/common/transformations/tests/common_optimizations/align_mixed_fp32_fp16_types_test.cpp

Comment (WeldonWangwang) [reply src/common/transformations/src/transformations/fp16_compression/align_mixed_fp32_fp16_types.cpp:69]: OK, will check this.

**Pull request closed without merging**